solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -201,10 +273,78 @@ pub mod solana_ibc {
     #[allow(unused_variables)]
     pub fn init_mint<'a, 'info>(
         ctx: Context<'a, 'a, 'a, 'info, InitMint<'info>>,
+        effective_decimals: u8,
         port_id: ibc::PortId,
         channel_id_on_b: ibc::ChannelId,
         hashed_base_denom: CryptoHash,
+        original_decimals: u8,
+        token_name: String,
+        token_symbol: String,
+        token_uri: String,
     ) -> Result<()> {
+        let private_storage = &mut ctx.accounts.storage;
+
+        if effective_decimals > original_decimals {
+            return Err(error!(error::Error::InvalidDecimals));
+        }
```

Comment:
With convert_decimals being able to handle conversion in whatever direction do we need this check?